agent/agent.go

```diff
@@ -21,13 +21,38 @@ import (
 	"github.com/google/gops/signal"
 )
 
-// Start starts the gops agent on a host process. Once agent started,
-// users can use the advanced gops features.
+// Agent represents an agent that enable the advanced gops features
+type Agent struct {
```

Comment:
The idea of exposing this type was for forward compatibility. It would allow the addition of further configuration attributes in the future without breaking the existing public interface.
Things like `gopsdir`, the listening address and even disabling certain events could be configured by `Agent` attributes in the future. There's even a TODO in the code regarding using HTTP instead of TCP directly, I'm not sure if that's the case but maybe this could also be configurable using this struct.